hazelcast-sql/src/main/java/com/hazelcast/sql/impl/expression/ExpressionEvalContextImpl.java

```diff
@@ -39,13 +46,23 @@ public class ExpressionEvalContextImpl implements ExpressionEvalContext {
     public ExpressionEvalContextImpl(
             @Nonnull List<Object> arguments,
             @Nonnull InternalSerializationService serializationService,
-            @Nonnull NodeEngine nodeEngine
-    ) {
+            @Nonnull NodeEngine nodeEngine) {
```

Comment:
some plans use this constructor (e.g. the ByKey plans), Subject should be mandatory for them in EEC, otherwise you could have query like `select get_ddl(inaccessible object) from my_imap where __key='exisiting'` return data to which user is not authorized

Reply:
Let's make it in a separate PR, please?

Reply:
why? This is simple - just use better constructor in a few more places. And maybe even remove the constructor without Subject.

Reply:
but if you really want we can do that (there should be also some tests with queries like I proposed), but for that you have to document in JIRA how the task is split into phases

Reply:
https://hazelcast.atlassian.net/browse/HZ-3128